src/core/execute.c

```diff
@@ -3251,20 +3252,17 @@ static int setup_smack(
                 r = mac_smack_apply_pid(0, context->smack_process_label);
                 if (r < 0)
                         return r;
-        }
-#ifdef SMACK_DEFAULT_PROCESS_LABEL
-        else {
+        } else if (manager->default_smack_process_label) {
                 _cleanup_free_ char *exec_label = NULL;
 
                 r = mac_smack_read_fd(executable_fd, SMACK_ATTR_EXEC, &exec_label);
                 if (r < 0 && !IN_SET(r, -ENODATA, -EOPNOTSUPP))
                         return r;
```

Comment:
This is called only when `DefaultSmackProcessLabel=` or compile-time default is specified. Is it expected?
Is it not necessary to read and apply the attribute even if the default is not specified??

IOW, why this block is conditionalized with `if (manager->default_smack_process_label)`?

Reply:
>This is called only when DefaultSmackProcessLabel= or compile-time default is specified.

Yes. This has been this way since e174dce271 ("smack: add default smack process label config").

> IOW, why this block is conditionalized with if (manager->default_smack_process_label)?

It is further explained in e419a0e310 ("core: set default process label only exec label is none"). I simply follow the logic that was before: the else block was executed if `SMACK_DEFAULT_PROCESS_LABEL` had been defined.